core/src/main/java/org/apache/calcite/sql/dialect/SparkSqlDialect.java

```diff
@@ -390,6 +391,30 @@ public SparkSqlDialect(SqlDialect.Context context) {
             timeUnitNode.getParserPosition());
         SqlFloorFunction.unparseDatetimeFunction(writer, call2, "DATE_TRUNC", false);
         break;
+      case COALESCE:
+        final SqlWriter.Frame coalesceFrame = writer.startFunCall("COALESCE");
+        List<SqlNode> operands = call.getOperandList();
+        int operandCount = operands.size();
+        for (int index = 0; index < operandCount; index++) {
+          SqlNode sqlNode = operands.get(index);
+          if (sqlNode instanceof SqlBasicCall && ((SqlBasicCall) sqlNode).getOperator()
+              .toString().equals("FORMAT")) {
+            final SqlWriter.Frame stringFrame = writer.startFunCall("STRING");
```

Comment:
This code can also be extracted to separate function (line 402-405) like unparseFormatForCoalesce()